### ISSUE TITLE
`adjust`/`adjust!` does not work with `ReactantOptimiser(OptimiserChain(...)`

### ISSUE THREAD

**Issue by benjwweber** [LuxDL/Lux.jl#1712]
I am trying to adjust the learning rate and weight decay of my optimiser, but found that `adjust!` does not do anything once the optimiser is moved to a device in the trainstate. This only fails if the optimiser is a `ReactantOptimiser` with `OptimiserChain`.

```julia-repl
julia> trainstate.optimizer
ReactantOptimiser(OptimiserChain(ReactantOptimiser(ClipGrad(delta=ConcreteIFRTNumber{Float32}(3.0f0))), ReactantOptimiser(AdamW(eta=ConcreteIFRTNumber{Float32}(0.001f0), beta=(ConcreteIFRTNumber{Float32}(0.9f0), ConcreteIFRTNumber{Float32}(0.999f0)), lambda=ConcreteIFRTNumber{Float32}(0.04f0), epsilon=ConcreteIFRTNumber{Float64}(1.0e-8), couple=true))))

julia> trainstate = Optimisers.adjust!(trainstate; eta = 100.0f0);

julia> trainstate.optimizer
ReactantOptimiser(OptimiserChain(ReactantOptimiser(ClipGrad(delta=ConcreteIFRTNumber{Float32}(3.0f0))), ReactantOptimiser(AdamW(eta=ConcreteIFRTNumber{Float32}(0.001f0), beta=(ConcreteIFRTNumber{Float32}(0.9f0), ConcreteIFRTNumber{Float32}(0.999f0)), lambda=ConcreteIFRTNumber{Float32}(0.04f0), epsilon=ConcreteIFRTNumber{Float64}(1.0e-8), couple=true))))

julia> Optimisers.adjust(trainstate.optimizer; eta = 100.0f0)
ReactantOptimiser(OptimiserChain(ReactantOptimiser(ClipGrad(delta=ConcreteIFRTNumber{Float32}(3.0f0))), ReactantOptimiser(AdamW(eta=ConcreteIFRTNumber{Float32}(0.001f0), beta=(ConcreteIFRTNumber{Float32}(0.9f0), ConcreteIFRTNumber{Float32}(0.999f0)), lambda=ConcreteIFRTNumber{Float32}(0.04f0), epsilon=ConcreteIFRTNumber{Float64}(1.0e-8), couple=true))))

julia> Optimisers.adjust(trainstate.optimizer.opt; eta = 100.0f0)
OptimiserChain(ReactantOptimiser(ClipGrad(delta=ConcreteIFRTNumber{Float32}(3.0f0))), ReactantOptimiser(AdamW(eta=ConcreteIFRTNumber{Float32}(100.0f0), beta=(ConcreteIFRTNumber{Float32}(0.9f0), ConcreteIFRTNumber{Float32}(0.999f0)), lambda=ConcreteIFRTNumber{Float32}(0.04f0), epsilon=ConcreteIFRTNumber{Float64}(1.0e-8), couple=true)))
```
